kq_epic_post.md

```diff
@@ -0,0 +1,160 @@
+
+Once we'd exhausted our best efforts, we ultimately decided to forgo this particular quest lest we awaken the Balrog. Due to the very small group of affected users, we determined that the estimated effort to continue was not worth the potential impact. Instead, there is an open discussion about updating our scorecard tooling to exempt these queries from our scorecards moving forward.
+
+#### Obstacle 3: The road goes ever on
+To populate our new `pull_requests.status`column, we needed to backfill with the data from issues.state. `pull_requests` is a _big_ table, so we knew we were in for a hefty process. What we didn't expect is that the transition would take 604 hours of run time across a period of 33 days, including dry runs. Within that, we had to maintain vigilance as we encountered unexplained failures that would interrupt the process, requiring us to manually restart. Finally, when we thought our toil had borne its reward, we were faced with some confusing data.
+
+While the logging indicated everything had run as expected, Kusto, datadot, and analytics hosts suggested the backfill had missed some pull requests[^2]. This was potentially an enormous setback. To find a reliable source of truth, we reached out to several teams and finally learned that what we needed was a new full snapshot of the table. With that done, we were able to determine that the backfill was mostly successful and the remaining group of pull requests missing `status` data were from the period after dual writing was introduced.
+
+#### Obstacle 4: But they were all of them deceived, for another bug was made
+And so we began fixing our dual writing bug, [a saga in itself](https://github.com/github/pull-requests/issues/15850). Our initial implementation set a pull request's status at the same time that an issue state was set via an existing ActiveRecord callback on the issue[^3]. The unexpected flaw with that was related to how an issue is created just before its associated pull request. Our first thought was to move this into its own callback on the pull request, but we decided to give orchestrations a try after some feedback from the Issues team[^4]. But we still encountered new mismatched records as there now a race condition between two orchestration processes and further iterations on this path caused deadlocks. Ultimately, the solution involved a callback on the issue that was localized to update pull request `status` whenever an issue's `state` changed. Phew.
```

Comment:
it might be worth describing the advantage orchestrations were thought to bring, or at least a link to a document explaining the background of orchestrations!

Reply:
I found a thread where an Issues team member briefly explained why they're pushing back on adding callbacks, but I couldn't find any docs about the Orchestrations pattern in that context.